doc/attr.xml

```diff
@@ -1310,3 +1310,42 @@ fail]]></Example>
   </Description>
 </ManSection>
 <#/GAPDoc>
+  
+<#GAPDoc Label="HamiltonianPath">
+<ManSection>
+  <Prop Name="HamiltonianPath" Arg="digraph"/>
+  <Returns>A list or <K>fail</K>.</Returns>
+  <Description>
+    Returns a Hamiltonian path if one exists, <K>fail</K> if not.
+
+    A <E>Hamiltonian path</E> is a directed circuit visiting
+    each vertex exactly once. See Section <Ref Subsect="Definitions"
+     Style="Number" /> for the definition of a directed circuit.
```

Comment:
You should actually read these definitions, since currently this documentation is not quite true. We define a directed circuit to be a particular kind sequence of vertices and edges. Your method just returns a sequence of vertices, so it's *not* returning a directed circuit, as you state. Please look at the documentation for `DigraphAllSimpleCircuits` to see an example of how to be more precise about this. Also - you should say `cycle` rather than `circuit` - again, look at the definition.

You could say something like `A <E>Hamiltonian path</E> of a digraph with <C>n</C> vertices is directed cycle of length <C>n</C>. If <A>digraph</A> is a digraph that contains a Hamiltonian path, then this function returns one, described in the form used by <Ref Attr="DigraphAllSimpleCircuits"/>...` and whatever else you need to make this bit of documentation correct and consistent.